py/objstringio.c

```diff
@@ -107,8 +107,19 @@ STATIC mp_uint_t stringio_ioctl(mp_obj_t o_in, mp_uint_t request, uintptr_t arg,
                     ref = o->vstr->len;
                     break;
             }
-            o->pos = ref + s->offset;
-            s->offset = o->pos;
+            new_pos = ref + s->offset;
+            if (s->offset < 0) {
```

Comment:
For SEEK_SET, it doesn't make sense to pass a negative offset, whereas it makes good sense to address a full unsigned range of offsets (e.g., being able to address full 4GB instead of meager 2GB). So, this if should include a check for it. This actually was the essence of my initial protests against this patch.